backend/src/auth/gestion-des-droits/roles/role.service.ts

```diff
@@ -0,0 +1,175 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { and, eq, sql } from 'drizzle-orm';
+import DatabaseService from '../../../common/services/database.service';
+import { AuthRole, AuthUser } from '../../models/auth.models';
+import {
+  utilisateurDroitTable,
+  UtilisateurDroitType,
+} from '@/backend/auth';
+import { NiveauAcces } from './niveau-acces.enum';
+import { utilisateurSupportTable } from './utilisateur-support.table';
+import { utilisateurVerifieTable } from './utilisateur-verifie.table';
+import { ResourceType } from '@/backend/auth/gestion-des-droits/resource-type.enum';
+import { Role } from '@/backend/auth/gestion-des-droits/roles/role.enum';
+import { dcpTable } from '@/backend/auth';
+
+
+@Injectable()
+export class RoleService {
+  private readonly logger = new Logger(RoleService.name);
+
+  constructor(private readonly databaseService: DatabaseService) {}
+
+  async getUserRolesForAResource(
+    user: AuthUser,
+    resourceType: ResourceType,
+    resourceId: number | null
+  ): Promise<Role[]> {
+    this.logger.log(
+      `Récupération des rôles de l'utilisateur ${user.id} sur la ressource ${resourceType} ${resourceId}`
+    );
+    const roles: Role[] = [];
+
+    if (user.role === AuthRole.AUTHENTICATED && user.id) {
+      // CONNECTE
+      roles.push(Role.CONNECTE);
+
+      // VERIFIE
+      const estVerifie = await this.estVerifie(user.id);
```

Comment:
Est-ce qu'on pourrait passer tous les `est`en `is`? Exemple : estVerifie devient `isVerifie`.